planning/motion_velocity_planner/autoware_motion_velocity_boundary_departure_prevention_module/src/utils.cpp

```diff
@@ -195,16 +207,24 @@ void update_departure_intervals(
   const trajectory::Trajectory<TrajectoryPoint> & aw_ref_traj, const double vehicle_length_m,
   const TrajectoryPoint & ref_traj_fr_pt, const double ego_dist_from_traj_front,
   const double th_pt_shift_dist_m, const double th_pt_shift_angle_rad,
-  const std::unordered_set<DepartureType> & enable_type)
+  const std::unordered_set<DepartureType> & enable_type, const bool is_reset_interval,
+  const bool is_departure_persist)
 {
   update_departure_intervals_poses(
     departure_intervals, aw_ref_traj, ref_traj_fr_pt, ego_dist_from_traj_front, th_pt_shift_dist_m,
     th_pt_shift_angle_rad);
 
   for (const auto side_key : g_side_keys) {
+    if (is_reset_interval) {
+      const auto remove_itr = std::remove_if(
+        departure_intervals.begin(), departure_intervals.end(),
+        [&](const DepartureInterval & interval) { return interval.side_key == side_key; });
+      departure_intervals.erase(remove_itr, departure_intervals.end());
+      continue;
+    }
     check_departure_points_between_intervals(
       departure_intervals, departure_points[side_key], aw_ref_traj, vehicle_length_m, side_key,
-      enable_type);
+      enable_type, is_departure_persist);
```

Comment:
rather than pass `is_departure_persist` to the function, it looks like a `if(is_departure_persist)` could be added before calling the function.